src/plugins/chart_expressions/expression_partition_vis/public/utils/filter_helpers.ts

```diff
@@ -48,20 +51,51 @@ export const getFilterClickData = (
     })
   );
 
+  const originalRowIndex = Math.floor(rowIndex / numOriginalMetrics);
+
   data.push(
-    ...clickedLayers.map((clickedLayer, index) => ({
-      column: visData.columns.findIndex((col) => col.id === bucketColumns[index].id),
-      row: matchingIndex,
-      value: clickedLayer.groupByRollup,
-      table: visData,
-    }))
+    ...(clickedLayers
+      .map((clickedLayer, index) => {
+        const currentColumn = visData.columns.find((col) => col.id === bucketColumns[index].id);
+
+        if (!currentColumn) {
+          return undefined;
+        }
+
+        const currentColumnIndex = visData.columns.findIndex((col) => col === currentColumn);
```

Comment:
nit, saves one n-complexity operation:
```suggestion
        const currentColumnIndex = visData.columns.findIndex((col) => col.id === bucketColumns[index].id);

        if (currentColumnIndex === -1) {
          return undefined;
        }

        const currentColumn = visData.columns[currentColumnIndex]
```

Reply:
Nice optimization